Makefile

```
@@ -0,0 +1,30 @@
+.PHONY: clean build lint docs
+
+clean: # clean all build, python, and testing files
+	rm -fr build/
+	rm -fr dist/
+	rm -fr .eggs/
+	find . -name '*.egg-info' -exec rm -fr {} +
+	find . -name '*.egg' -exec rm -f {} +
+	find . -name '*.pyc' -exec rm -f {} +
+	find . -name '*.pyo' -exec rm -f {} +
+	find . -name '*~' -exec rm -f {} +
+	find . -name '__pycache__' -exec rm -fr {} +
+	rm -fr .tox/
+	rm -fr .coverage
+	rm -fr coverage.xml
+	rm -fr htmlcov/
+	rm -fr .pytest_cache
+	rm -fr .mypy_cache
+
+build: # run tests
+	pytest
+
+lint: # run formatting and linting
+	black -l 88 .
+	isort -l 88 .
+	ruff .
```

One of the warnings on running the linter involved updating how `ruff` is called.
```suggestion
	ruff check ./subcell_analysis
```